core/iwasm/common/wasm_runtime_common.c

```diff
@@ -2005,15 +1978,10 @@ wasm_runtime_call_wasm(WASMExecEnv *exec_env,
                                 param_argc, new_argv);
 #endif
     if (!ret) {
-        if (clear_wasi_proc_exit_exception(exec_env->module_inst)) {
-            ret = true;
-        }
-        else {
-            if (new_argv != argv) {
-                wasm_runtime_free(new_argv);
-            }
-            return false;
+        if (new_argv != argv) {
+            wasm_runtime_free(new_argv);
```

Comment:
`clear_wasi_proc_exit_exception` is only applied after calling wasi `_start` function now, is it a general handling of wasi? Will developer call other functions in wasi mode and then `wasm_proc_exit` occurs?

Reply:
i guess it's a gray area.
proc_exit is supposed to terminate the "process".
the concept of "process" only makes sense in the context of a wasi command.

Reply:
Sounds reasonable, maybe we can apply the changes first, and modify if needed in the future.